src/services/expensesService.js

```diff
@@ -0,0 +1,69 @@
+const { data } = require('../data');
+
+const getAllExpenses = () => {
+  return data.expenses;
+};
+
+const getById = (id) => {
+  return data.expenses.find((exp) => exp.id === parseInt(id));
+};
+
+const normalizeCategory = (categories) => {
+  return Array.isArray(categories) || !categories ? categories : [categories];
+};
+
+const filterExpensesById = (id) => {
+  data.expenses = data.expenses.filter((exp) => exp.userId === parseInt(id));
+
+  return data.expenses;
+};
+
+const filterExpensesByCategory = (category) => {
+  return data.expenses.filter((exp) => category.includes(exp.category));
+};
+
+const filterExpensesByDate = (from, to) => {
+  return data.expenses.filter((exp) => {
+    const spentAt = new Date(exp.spentAt);
+
+    return spentAt >= from && spentAt <= to;
+  });
+};
+
+const create = ({ userId, spentAt, title, amount, category, note }) => {
+  const updatedExpense = {
```

Comment:
Code quality issue: The `create` function should check if the provided parameters are valid. For example, it should check if `userId`, `spentAt`, `title`, `amount`, `category`, and `note` are not `undefined` or `null`.